docs/source/examples/ports.rst

```diff
@@ -0,0 +1,71 @@
+.. _ports:
+
+Opening Ports
+=============
+
+At times, it might be crucial to expose specific ports on your cluster to the public internet. For example:
+
+- **Exposing Development Tools**: If you're working with tools like Jupyter Notebook or ray, you'll need to expose its port to access the interface / dashboard from your browser.
+- **Creating Web Services**: Whether you're setting up a web server, database, or another service, they all communicate via specific ports that need to be accessible.
+- **Collaborative Tools**: Some tools and platforms may require port openings to enable collaboration with teammates or to integrate with other services.
+
+Opening Ports for SkyPilot cluster
+----------------------------------
+
+SkyPilot makes the process of opening ports simple. Here's a basic example in YAML configuration to expose a Jupyter Lab server:
+
+.. code-block:: yaml
+
+    # jupyter_lab.yaml
+    resources:
+      ports: 8888
+
+    setup: pip install jupyter
+
+    run: jupyter lab --port 8888 --no-browser --ip=0.0.0.0
+
+In this example, by specifying :code:`ports` in the :code:`resources` section, SkyPilot will automatically open port 8888 on your cluster. The :code:`run` command will start the Jupyter Lab server on port 8888. To access the server, run:
+
+.. code-block:: bash
+
+    $ sky launch -c jupyter jupyter_lab.yaml
+
+and look for the logs fo some output like:
+
+.. code-block:: bash
+
+    Jupyter Server 2.7.0 is running at:
+        http://127.0.0.1:8888/lab?token=<token>
+
+Run
+
+.. code-block:: bash
+
+    $ sky status -a jupyter
```

Comment:
We need to get #2439. The output of `sky status -a` can be quite overwhelming (too long).

Reply:
sure 🫡 I can work on that